common/json.c

```diff
@@ -471,6 +471,15 @@ void json_add_hex(struct json_result *result, const char *fieldname,
 	json_add_string(result, fieldname, hex);
 }
 
+void json_add_hex_reversed(struct json_result *result, const char *fieldname,
+		  const void *data, size_t len)
+{
+	char hex[hex_str_size(len)];
+
+	hex_encode_reversed(data, len, hex, sizeof(hex));
+	json_add_string(result, fieldname, hex);
+}
+
```

Comment:
I prefer creating json_add_txid() and json_add_blkhash() which call this underneath.  That way we're not relying on the callers to know they have to use this routine instead of json_add_hex().

Eventually we should acreate wrapper types for txids and blockids, then implement type_to_string for them, which also fixes the fact that they're backwards in the logs.  But it's a big job, so let's start small?